### PR TITLE
[IT-3145] Add coverage to pytest

### PR DESCRIPTION
**Pull request by jesusaurus**
Run our unit tests with coverage and upload the results to coveralls.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -26,7 +26,9 @@ jobs:
           python-version: 3.9
       - run: pip install -U pipenv
       - run: pipenv install --dev
-      - run: pipenv run python3 -m pytest tests/ -vv
+      - run: pipenv run coverage run -m pytest tests/ -vv
+      - name: upload coverage to coveralls
+        uses: coverallsapp/github-action@v2
```
Comment (zaro0508) [inline .github/workflows/test.yaml:31]: doesn't this need a token to upload to the coverall service?

Comment (jesusaurus) [reply .github/workflows/test.yaml:31]: I think it's using the default github token available in github actions, but I'm also surprised that this Just Works(tm). 

The [documentation](https://github.com/marketplace/actions/coveralls-github-action) is a little confusing, the `github-token` input is listed as `required` but the description also specifies a default value and the examples don't set it. And you can see the results of this PR run at https://coveralls.io/builds/62919609

Comment (zaro0508) [reply .github/workflows/test.yaml:31]: i'm guessing that GHA was, some how, enabled with OIDC access to the coveralls service.  It's also probably enabled at the organization level. 

web page looks nice, but it shows coverage for `test` folder, shouldn't that be excluded?

Comment (jesusaurus) [reply .github/workflows/test.yaml:31]: oh good point, yeah `tests/` should probably be excluded.

**Pull request merged**